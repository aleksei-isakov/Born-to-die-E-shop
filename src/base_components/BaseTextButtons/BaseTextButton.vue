<template>
  <BaseButton class="base-text-button" @click="onClickEmitEvent">
    <slot></slot>
  </BaseButton>
</template>

<script>
import { BaseButton } from '@/base_components/';

export default {
  name: 'BaseTextButton',

  components: {
    BaseButton
  },

  methods: {
    onClickEmitEvent() {
      this.$emit('click');
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';
.base-text-button {
  height: fit-content;
  padding: 7px 17px;
  border: 0;
  border-radius: 4px;
  text-transform: uppercase;
  font-weight: 500;
  font-size: 14px;
  line-height: 22px;
  letter-spacing: 0.03em;
  font-family: Roboto, sans-serif;
  transition: $transition;
  cursor: pointer;
}
</style>
