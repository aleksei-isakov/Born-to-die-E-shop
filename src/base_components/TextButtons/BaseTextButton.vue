<template>
  <BaseButton class="base-text-button" @click="onClickEmitEvent">
    <slot></slot>
  </BaseButton>
</template>

<script>
import { BaseButton } from '..';

export default {
  name: 'BaseTextButton',

  components: {
    BaseButton
  },

  methods: {
    onClickEmitEvent() {
      this.$emit('click');
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/variables.scss';
.base-text-button {
  height: fit-content;
  padding: 9px 17px;
  border: 0;
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
  font-size: 14px;
  line-height: 22px;
  letter-spacing: 0.03em;
  font-family: Roboto, sans-serif;
  transition: $transition;
  cursor: pointer;
}
</style>
