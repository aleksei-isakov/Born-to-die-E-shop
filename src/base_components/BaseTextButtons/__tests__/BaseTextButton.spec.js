import { shallowMount, createLocalVue } from '@vue/test-utils';
import BaseTextButton from '../BaseTextButton.vue';

describe('BaseTextButton.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue();

    wrapper = shallowMount(BaseTextButton, {
      localVue
    });

    wrapper.vm.onClickEmitEvent();
    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.is(BaseTextButton)).toBeTruthy();
  });

  it('renders a valid snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
