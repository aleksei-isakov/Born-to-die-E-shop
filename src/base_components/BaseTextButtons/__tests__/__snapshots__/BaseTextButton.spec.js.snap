// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTextButton.vue renders a valid snapshot 1`] = `
<basebutton-stub
  class="base-base-text-button"
/>
`;
